### PR TITLE
Avoid deleting ACLs when editing

### PR DESCRIPTION
**Pull request by jvorcak**
Fixes https://github.com/redpanda-data/console/issues/1220

Please try to test it yourself as well, I tested on my setup, but I'd appreciate if somebody could double-check this.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
         "array-move": "^4",
         "framer-motion": "^7",
         "js-base64": "^3.7.5",
+        "lodash-es": "^4.17.21",
```
Comment (rikimaru0345) [inline frontend/package.json:42]: Not worth adding a dependency for just isEqual.

There are 3 options that are better:

You can either iterate over `Object.entries` and compare property by property, or use mobx `comparer.structural` (which essentially does the same thing, but also nesting), or simply write a custom isEqual function,

**Review by rikimaru0345 — CHANGES_REQUESTED** [review]

**Comment by rikimaru0345** [comment]
How does this prevent that ACLs that existed before (and don't need to be modified) are not created again? or is that not an issue? if not, why?

**Comment by jvorcak** [comment]
If they don't need to be modified, we remote then from `allToCreate` array thus they're not created.

File: frontend/package.json (hunk 1 above)
Comment (jvorcak) [reply frontend/package.json]: @rikimaru0345 done

File: frontend/src/components/pages/acls/PrincipalGroupEditor.tsx [hunk 2]
```diff
@@ -42,22 +51,29 @@ export const AclPrincipalGroupEditor = observer((p: {
...
-                else throw new Error('No targeted resources. You can delete this ACL group from the list view.');
+            if (allToCreate.length == 0) {
+                if (p.type == 'create') {
+                    throw new Error('Creating an ACL group requires at least one resource to be targeted. Topic/Group targets with an empty selector are not valid.');
+                } else {
+                    throw new Error('No targeted resources. You can delete this ACL group from the list view.');
+                }
+            }
 
             // Delete all ACLs in group
```
Comment (rikimaru0345) [inline frontend/src/components/pages/acls/PrincipalGroupEditor.tsx:62]:
```suggestion
            // Ignore creation of ACLs that already exist, and delete ACLs that are no longer needed
```

File: frontend/src/components/pages/acls/PrincipalGroupEditor.tsx [hunk 3]
```diff
@@ -42,22 +51,29 @@ export const AclPrincipalGroupEditor = observer((p: {
...
-                        host: group.host,
-                        operation: 'Any',
-                        permissionType: 'Any'
-                    });
+                if (group.sourceEntries.length > 0) {
+                    const requests = group.sourceEntries.map(acl => {
+                        // try to find this in allToCreate
+                        const foundIdx = allToCreate.findIndex(x => comparer.structural(acl, x))
+                        if(foundIdx !== -1) {
+                            allToCreate.splice(foundIdx, 1)
```
Comment (rikimaru0345) [inline frontend/src/components/pages/acls/PrincipalGroupEditor.tsx:69]:
```suggestion
                            // acl already exists, remove it from the list
                            allToCreate.splice(foundIdx, 1)
```

File: frontend/src/components/pages/acls/PrincipalGroupEditor.tsx [hunk 4]
```diff
@@ -42,22 +51,29 @@ export const AclPrincipalGroupEditor = observer((p: {
...
+                            return Promise.resolve()
+                        } else {
+                            return api.deleteACLs(acl)
```
Comment (rikimaru0345) [inline frontend/src/components/pages/acls/PrincipalGroupEditor.tsx:72]:
```suggestion
                            // acl should no longer exist, delete it
                            return api.deleteACLs(acl)
```

**Review by rikimaru0345 — APPROVED** [review]

**Pull request merged**